### ISSUE TITLE
[9.0.100-alpha.1.24055.6] WPF uses WinForm PropertyGrid Tooltip get "Adding the tip to the native ToolTip control did not succeed." exception

### ISSUE THREAD

**Issue by Junjun-zhao** [dotnet/winforms#10614]
### .NET version

```
.NET SDK:
 Version:           9.0.100-alpha.1.24055.6
 Commit:            ca1439c8ae
 Workload version:  9.0.100-manifests.f3b52893

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.19044
 OS Platform: Windows
 RID:         win-x64
 Base Path:   C:\Program Files\dotnet\sdk\9.0.100-alpha.1.24055.6\

.NET workloads installed:
 There are no installed workloads to display


Host:
  Version:      9.0.0-alpha.1.24054.18
  Architecture: x64
  Commit:       9325d3836d

.NET SDKs installed:
    9.0.100-alpha.1.24055.6 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 9.0.0-alpha.1.24055.3 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 9.0.0-alpha.1.24054.18 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 9.0.0-alpha.1.24055.1 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
```

### Did it work in .NET Framework?

Yes

### Did it work in any of the earlier releases of .NET Core or .NET 5+?

Yes
1. Windows10 21h2 x64 + dotnet-sdk-7.0.405: Pass
2. Windows10 21h2 x64 + dotnet-sdk-8.0.101: Pass
3. Windows10 21h2 x64 + dotnet-sdk-9.0.100-alpha.1.24055.6: Fail

### Issue description

When using WinForm PropertyGrid control in WPF app, the tooltip in PropertyGrid will throw "System.InvalidOperationException: Adding the tip to the native ToolTip control did not succeed." exception.

**Application Name**: AvalonEdit
**OS**： Windows 10 21H2
**CPU**: X64
**.NET Build Number**: dotnet-sdk-9.0.100-alpha.1.24055.6
**App or App source checking at:** https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1936985
**Github Link:** https://github.com/icsharpcode/AvalonEdit

### Steps to reproduce

**Minimal Repro steps** (Demo attached [TooltipDemo.zip](https://github.com/dotnet/winforms/files/13871438/TooltipDemo.zip)):
**The machine has dotnet-sdk-9.0.100-alpha.1.24055.6 installed.**

​​1.Create a .NET 7 or .NET 8 WPF project.
2. Add ```<UseWindowsForms>true</UseWindowsForms> ```to project file:
```
 <PropertyGroup>
   <OutputType>WinExe</OutputType>
   <TargetFramework>net8.0-windows</TargetFramework>
   <Nullable>enable</Nullable>
   <ImplicitUsings>enable</ImplicitUsings>
   <UseWPF>true</UseWPF>
   <UseWindowsForms>true</UseWindowsForms>
 </PropertyGroup>
```
3. Add PropertyGrid control to MainWindow.xaml:
```
<Window x:Class="ToolTipsDemo.MainWindow"
        xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
        xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
        xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
        xmlns:local="clr-namespace:ToolTipsDemo"
        xmlns:forms="clr-namespace:System.Windows.Forms;assembly=System.Windows.Forms"
        mc:Ignorable="d"
        Title="MainWindow" Height="450" Width="800">
    <Grid>
        <WindowsFormsHost Name="propertyGridHost">
            <forms:PropertyGrid x:Name="propertyGrid"/>
        </WindowsFormsHost>
       
    </Grid>
</Window>
```
4. Build the project.
5. Update the runtime config file to .NET 9 versions:
```
"frameworks": [
      {
        "name": "Microsoft.NETCore.App",
        "version": "9.0.0-alpha.1.24054.18"
      },
      {
        "name": "Microsoft.WindowsDesktop.App",
        "version": "9.0.0-alpha.1.24055.1"
      }
    ],
```
6. Launch the app.
7. Move the mouse to "Alphabetical" button of PropertyGrid.

**Expected Result:**
Tooltip for "Alphabetical" button will appear.

**Actual Result:**
Exception window pops up:
![image](https://github.com/dotnet/winforms/assets/83067197/f6c351e7-2712-495d-8594-540ef76d767d)

**Findings:**

In WPF project, any controls that has tooltip in PropertyGrid will cause this exception.
In WPF project, default WPF controls, like a button, when adding tooltip for them, they are working as normal.
In WinForm project, the tooltip in PropertyGrid doesn't have this issue.

StackTrace:
```
System.InvalidOperationException: Adding the tip to the native ToolTip control did not succeed.
   at System.Windows.Forms.ToolTip.SetToolInfo(Control control, String caption)
   at System.Windows.Forms.ToolTip.Hide(IWin32Window win)
   at System.Windows.Forms.ToolStrip.UpdateToolTip(ToolStripItem item, Boolean refresh)
   at System.Windows.Forms.ToolStripItem.HandleMouseLeave(EventArgs e)
   at System.Windows.Forms.ToolStrip.HandleMouseLeave()
   at System.Windows.Forms.ToolStrip.OnMouseLeave(EventArgs e)
   at System.Windows.Forms.Control.WndProc(Message& m)
   at System.Windows.Forms.ToolStrip.WndProc(Message& m)
   at System.Windows.Forms.NativeWindow.Callback(HWND hWnd, MessageId msg, WPARAM wparam, LPARAM lparam)
```
@dotnet-actwx-bot @dotnet/compat

**Comment by Tanya-Solyanik** [comment]
@LeafShi1 - had you tried reverting that PR to see if the issue is gone?

**Comment by LeafShi1** [comment]
> @LeafShi1 - had you tried reverting that PR to see if the issue is gone?

Yes, if revert the PR https://github.com/dotnet/winforms/pull/9900, the issue is gone.

**Comment by Tanya-Solyanik** [comment]
Oh, I see, there are changes in GridToolTip in that PR. Thank you!

**Comment by Junjun-zhao** [comment]
@Tanya-Solyanik Any update for this issue? Could you please confirm whether this is blocker for .NET 9 Preview 1?

**Comment by merriemcgaw** [comment]
@Junjun-zhao I don't think this is a blocker for Preview 1, but we should certainly get something figured out for Preview 2. @Tanya-Solyanik, @JeremyKuhne do you agree?

**Comment by Zheng-Li01** [comment]
Verified this issue on .NET 9.0.0-preview.2.24079.1 with dlls built from winforms repo of main branch, the issue has been fixed that the tooltip appeared when moving the mouse to "Alphabetical" button of PropertyGrid.

![image](https://github.com/dotnet/winforms/assets/38325459/7597ffa1-0210-4eee-9a2b-2c823588f631)

**Comment by Junjun-zhao** [comment]
Verified and this issue has been fixed with dotnet-sdk-9.0.100-preview.2.24113.6

**Comment by Syareel-Sukeri** [comment]
Verified this issue with .NET SDK 9.0 Preview 2 test pass build, this issue was fixed. Test result is the same as above where tooltip will appear when hover to the "Alphabetical" button.
![image](https://github.com/dotnet/winforms/assets/156630333/b6243d9e-c095-4a66-82ad-b074217f5594)
